tests/src/JIT/Directed/arglist/vararg.cs

```diff
@@ -0,0 +1,5027 @@
+            {
+                sum += expectedFloatValues[i];
+            }
+
+            for (int i = 0; i < expectedDoubleValues.Length; ++i)
+            {
+                sum += expectedDoubleValues[i];
+            }
+
+            return sum == expectedSum;
+        }
+
+        /// <summary>
+        /// Given an input set create an arglist to pass to a vararg callee.
+        /// 
+        /// The callee will simply loop over the arguments, compute the sum
+        /// then return the value.
+        /// 
+        /// Do a quick check on the value returned, and return whether they 
+        /// are equal.
+        /// 
+        /// </summary>
+        /// <param name="expectedValues"></param>
+        /// <returns>bool</returns>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool TestPassingIntsAndFloatsManaged()
+        {
+            int a = 10;
+            int b = 20;
+            int c = 30;
+
+            float f1 = 1.0f;
+            float f2 = 2.0f;
+            float f3 = 3.0f;
+
+            float expectedSum = 0.0f;
+
+            expectedSum = a + b + c + f1 + f2 + f3;
+
+            float calculatedSum = ManagedNativeVarargTests.TestPassingIntsAndFloats(
+                expectedSum,
+                __arglist(
+                    a,
+                    f1,
+                    b,
+                    f2,
+                    c,
+                    f3
+                )
+            );
+
+            return expectedSum == calculatedSum;
+        }
+
+        /// <summary>
+        /// Given an input set create an arglist to pass to a vararg callee.
+        /// 
+        /// The callee will simply loop over the arguments, compute the sum
+        /// then return the value.
+        /// 
+        /// Do a quick check on the value returned, and return whether they 
+        /// are equal.
+        /// 
+        /// </summary>
+        /// <param name="expectedValues"></param>
+        /// <returns>bool</returns>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool TestPassingLongsAndDoublesManaged()
+        {
+            long[] expectedIntValues = new long[] { 10, 20, 30 };
+            double[] expectedFloatValues = new double[] { 1.0, 2.0, 3.0 };
+
+            double expectedSum = 0.0f;
+
+            for (int i = 0; i < expectedIntValues.Length; ++i) expectedSum += expectedIntValues[i];
+            for (int i = 0; i < expectedFloatValues.Length; ++i) expectedSum += expectedFloatValues[i];
+
+
+            double calculatedSum = ManagedNativeVarargTests.TestPassingLongsAndDoubles(
+                expectedSum,
+                __arglist(
+                    expectedIntValues[0],
+                    expectedFloatValues[0],
+                    expectedIntValues[1],
+                    expectedFloatValues[1],
+                    expectedIntValues[2],
+                    expectedFloatValues[2]
+                )
+            );
+
+            return expec
```

Comment:
It looks like the usage here is: return as the test result the id# of the last test that failed, which better be something besides 100. Seems like you should instead have:
```
if (m_failCount == 0)
{
    return 100;
}
else
{
    return 101; // some failure code
}
```
and have the ReportFailure functions output the test id along with the failure message.

Because the test id# is not sequential, it's hard to know if you have a dense, non-overlapping set of id#s.